### ISSUE TITLE
[Bug] Lib modules fails to find suffix for Ubuntu 14 LTS

### ISSUE THREAD

**Issue by Kulgar** [jarib/ffi-icu#29]
Hi there :)

Under Ubuntu 14 LTS (and I'm sure it also happens in 12 LTS), the gem fails to find the right suffix. 

Under Ubuntu here is the printed version for ICU lib:

```ruby
2.1.3 :003 > ICU::Lib.version.to_s
 => "52.1" 
```

Now if you do: 

```ruby
2.1.3 :013 > ICU::Lib.figure_suffix(ICU::Lib.version.to_s)
 => nil 
```

You don't have the right suffix. But if you do: 

```ruby
2.1.3 :006 > ICU::Lib.figure_suffix("52")
 => "_52" 
```

You have it. Looking at:  https://github.com/jarib/ffi-icu/blob/master/lib/ffi-icu/lib.rb#L102

You see that the "dot" is not handled at all. So I think we should add something like: 

suffixes << "_#{version}" << "_#{version[0].chr}_#{version[1].chr}" << "_#{version.split('.')[0]}" 

Do you think it would solve the issue? Do you want me to do a pull request after testing this out?
This bug is quite important for me as it breaks some functionnalities in applications using the gem. :)

Thanks!
Regis.

**Comment by jarib** [comment]
@cfis Any opinions on this and the associated PR? I think you wrote the code in question here.

**Comment by jarib** [comment]
Btw, all the specs are passing on Ubuntu 14.04.1 and 14.04.3 for for me, so not really sure what's up here.

**Comment by Kulgar** [comment]
Hmm, but there are nothing that is actually testing the figure_suffix in rspec tests. 

There should be a test that fails if `ICU::Lib.figure_suffix(ICU::Lib.version.to_s)` returns nil explaining that lib icu might not be installed if ICU::Lib.version.to_s returns nil and printing version if it is installed.

**Comment by cfis** [comment]
I don't really have an opinion outside of its really annoying libicu does this.   Not surprised its fragile.

**Comment by jarib** [comment]
@Kulgar Even though `figure_suffix` isn't explicitly tested, all the specs would fail if it got the function names wrong.

I guess it doesn't hurt to add another suffix to the algorithm, so I'll merge this.

**Comment by jarib** [comment]
Released as 0.1.10

**Comment by Kulgar** [comment]
Hmm, I've just checked the rspec, it seems that you are not testing what is failing in the app I use. 

I added this rspec test: 

```ruby
    it "Test attach function" do
      suffix = ICU::Lib.figure_suffix(ICU::Lib.version.to_s)

      ICU::Lib.attach_function(:ucol_getRules, "ucol_getRules#{suffix}", [:pointer, :pointer], :pointer)
      ICU::Lib.attach_function(:ucol_getSortKey, "ucol_getSortKey#{suffix}", [:pointer, :pointer, :int, :pointer, :int], :int)
      ICU::Lib.attach_function(:ucol_getAttribute, "ucol_getAttribute#{suffix}", [:pointer, :int, :pointer], :int)
      ICU::Lib.attach_function(:ucol_setAttribute, "ucol_setAttribute#{suffix}", [:pointer, :int, :int, :pointer], :void)
    end
```

Those 4 lines are the ones included in the code of Canvas LMS app. If I write this in rspec, the test fails if I comment out my fix. 

In the light of these elements, would you know why no rspec tests are failing (and why those are)?
